core/src/main/java/hudson/model/Computer.java

```diff
@@ -990,14 +990,13 @@ public final long getDemandStartMilliseconds() {
      * Called by {@link Executor} to kill excessive executors from this computer.
      */
     /*package*/ void removeExecutor(final Executor e) {
-        Queue.withLock(new Runnable() {
+        final Runnable task = new Runnable() {
             @Override
             public void run() {
                 synchronized (Computer.this) {
                     executors.remove(e);
                     addNewExecutorIfNecessary();
-                    if (!isAlive()) // TODO except from interrupt/doYank this is called while the executor still isActive(), so how could !this.isAlive()?
```

Comment:
Yes as this call can now happen on the background thread so it may now be not alive when the thread scheduler gets around to scheduling execution